nirps/apero_raw_data_check/apero_raw_tests/tests/eng_test.py

```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Blank example test do not modify this.
+
+Copy this to create your test
+
+Created on 2023-07-03 at 14:37
+
+@author: cook
+"""
+from typing import Any, Dict
+
+from apero_raw_tests.core import io
+from apero_raw_tests.core import misc
+from astropy.io import fits
+import os
+import glob
+import numpy as np
+from astropy.table import Table
+import copy
+
+# =============================================================================
+# Define variables
+# =============================================================================
+# name of the column in database, must be unique
+NAME = 'ENG_TEST'
+# define any other constants here that you want moving to the parameters.py
+#  file (these can be overwritten by the yaml file) and may change
+#  depending on the profile used (i.e. NIRPS_HA or NIRPS_HE)
+
+
+# =============================================================================
+# Define functions
+# =============================================================================
+def test(params: Dict[str, Any], obsdir: str, log=False) -> bool:
+    """
+    Test with observation directory exists
+
+    Passed = True
+
+    :param params: dictionary of parameters (from yaml/cmd line/ parameters.py)
+    :param obsdir: str, the observation directory (e.g. the night directory)
+    :param log: bool, if True prints messages (all messages must be wrapped
+                in a if log: statement)
+
+    :return: bool, True if passed, False otherwise
+    """
+    # define parameters we use here
+    raw_directory = params['raw dir']
+    # -------------------------------------------------------------------------
+    if log:
+        msg = 'Analysing observation directory: {0}'
+        margs = [obsdir]
+        misc.log_msg(msg.format(*margs), level='')
+    # -------------------------------------------------------------------------
+    # get all observation directories
```

Comment:
This may take a while (and will increase with number of directories) this was a test to find whether we have a valid directory and does not need to be repeated.

Instead of:
```
    # get all observation directories
    obsdirs = io.get_obs_dirs(params)
    # -------------------------------------------------------------------------
    # test if observation directory exists in our list
    if obsdir not in obsdirs:
        if log:
            msg = ('OBSDIR TEST: Observation directory {0} does not exist in '
                   '{1} - TEST FAILED')
            margs = [obsdir, raw_directory]
            misc.log_msg(msg.format(*margs), level='warning')
        return False
```

I would define the obsdir absolute path:
```
obsdir_path = os.path.join(raw_directory, obsdir)
```
then test that whether it exists - if it doesn't it is enough to return False here
Remember to also have a log statement
i.e.
```
if not os.path.exists(obsdir_path):
    if log: 
        print('Observation directory {0} does not exist - TEST FAILED')
    return False
```

Reply:
done